src/OrbitGl/GraphTrack.cpp

```diff
@@ -31,19 +35,18 @@ GraphTrack<Dimension>::GraphTrack(CaptureViewElement* parent, TimeGraph* time_gr
 
 template <size_t Dimension>
 float GraphTrack<Dimension>::GetHeight() const {
-  float scale_factor = IsCollapsed() ? 1 : Dimension;
-  float height = layout_->GetTrackTabHeight() + GetLegendHeight() +
-                 layout_->GetTextBoxHeight() * scale_factor +
-                 layout_->GetSpaceBetweenTracksAndThread() + layout_->GetEventTrackHeightFromTid() +
-                 layout_->GetTrackBottomMargin();
+  float height = layout_->GetTrackTabHeight() + layout_->GetTrackContentTopMargin() * 2 +
+                 GetLegendHeight() + GetGraphContentHeight() +
+                 layout_->GetTrackContentBottomMargin();
   return height;
 }
 
 template <size_t Dimension>
 float GraphTrack<Dimension>::GetLegendHeight() const {
-  bool has_legend = !IsCollapsed() && Dimension > 1;
-  float legend_height = has_legend ? layout_->GetTextBoxHeight() : 0;
-  return legend_height;
+  if (IsCollapsed() || Dimension <= 1) {
+    return 0;
+  }
+  return layout_->GetTextBoxHeight() / 2.f;
```

Comment:
Not clear why you are dividing by 2 here. I know that maybe this was before in other part, but consider to add a comment.

Reply:
AFAIK it was like this before and it's a bit arbitrary... I've added a comment. IMO the main idea is that this space *should* scale with the overall size of texts in Orbit, but it shouldn't be as large as other text boxes, and introducing a new layout parameter just for this feels overkill.